torch/ao/quantization/fx/_model_report/_detector.py

```diff
@@ -134,9 +119,151 @@ def _detect_per_channel_helper(model: nn.Module):
+                pre_obs_dist_classification = DEFAULT_NON_STATIONARY
+                post_obs_dist_classification = DEFAULT_STATIONARY
+            elif pre_stat <= tolerance and post_stat <= tolerance:
+                dynamic_recommended = True  # dynamic best if input, output non-stationary
+                pre_obs_dist_classification = DEFAULT_NON_STATIONARY
+                post_obs_dist_classification = DEFAULT_NON_STATIONARY
+            elif pre_stat > tolerance and post_stat <= tolerance:
+                dynamic_recommended = True  # dynamic best if input stationary, output non-stationary
+                pre_obs_dist_classification = DEFAULT_STATIONARY
+                post_obs_dist_classification = DEFAULT_NON_STATIONARY
+            else:
+                raise Exception("Should always take one of above branches")
+
+            # store the set of important information for this module
+            module_info = {
+                "tolerance": tolerance,
+                "dynamic_recommended": dynamic_recommended,
+                "pre_observer_comp_stat": pre_stat,
+                "pre_observer_data_dist": pre_obs_dist_classification,
+                "post_observer_comp_stat": post_stat,
+                "post_observer_data_dist": post_obs_dist_classification,
+            }
+
+            module_dynamic_static_info[name] = module_info
+
+    dynamic_vs_static_string = "Dynamic vs. Static Quantization suggestions: \n"
+
+    for module_name in module_dynamic_static_info.keys():
+
+        module_info = module_dynamic_static_info[module_name]
+        suggestion_string_template = "For module {} it is suggested to use {} quantization because {}.\n"
+
+        # decide what string formatting values will be
+        quantization_type = ""
+
+        # TODO CHANGE TO NON STATIONARY AND STATIONARY TALK
+        quantization_reasoning = "the distribution of data before {} is {} and the distribution after is {}."
+        dynamic_benefit = " You will get more accurate results if you use dynamic quantization"
+        static_benefit = " You can increase model efficiency if you use static quantization"
+
+        if module_info["dynamic_recommended"]:
+            quantization_type = "dynamic"
+            quantization_reasoning = (
+                quantization_reasoning.format(
+                    module_name, module_info["pre_observer_data_dist"], module_info["post_observer_data_dist"]
+                )
+                + dynamic_benefit
+            )
+        else:
+            quantization_type = "static"
+            quantization_reasoning = (
+                quantization_reasoning.format(
+                    module_name, module_info["pre_observer_data_dist"], module_info["post_observer_data_dist"]
+                )
+                + static_benefit
+            )
+
+        # if we have a non-stationary input -> linear -> stationary input we suggest
+        if (
```

Comment:
can you add more details here? looks like we are making some recommendations here again, can we separate the code that we calculate the recommendation and the code that is used for printing the recommendation instead of mixing them?